src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -609,26 +609,31 @@ public ResultWrapper<IEnumerable<object>> eth_getFilterChanges(UInt256 filterId)
 
     public ResultWrapper<IEnumerable<FilterLog>> eth_getFilterLogs(UInt256 filterId)
     {
-        int id = (int)filterId;
+        CancellationTokenSource cancellationTokenSource = new(_rpcConfig.Timeout);
+        CancellationToken cancellationToken = cancellationTokenSource.Token;
 
-        return _blockchainBridge.FilterExists(id)
-            ? ResultWrapper<IEnumerable<FilterLog>>.Success(_blockchainBridge.GetFilterLogs(id))
-            : ResultWrapper<IEnumerable<FilterLog>>.Fail($"Filter with id: '{filterId}' does not exist.");
-    }
+        int id = (int)filterId;
```

Comment:
What would happen on overflow? Shall we support bigger filter id's?

Reply:
User can put anything here

Reply:
Filter id generation seed is int, so added a check before cast to avoid overflow.